lib/puppet/util/windows/api_types.rb

```diff
@@ -33,6 +33,22 @@ def self.from_string_to_wide_string(str, &block)
       nil
     end
 
+    def self.from_string_to_secure_wide_string(str, &block)
```

Comment:
I wonder if this is ultimately a little pointless given copies of the same strings floating around in the Ruby managed heap?